# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl
# Commit: 459739221a88
# Full Hash: 459739221a884b2b1c2adab42fd19fafe279bd20
# Author: Sandor Molnar <smolnar@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing build bustages
#   
#   This reverts commit ab681e36aeafc21ba62bb4680bdaa6206f02546d.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r fc9986230203 -r 459739221a88 toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl
--- a/toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl	Thu Jun 05 00:15:19 2025 +0300
+++ b/toolkit/components/credentialmanagement/nsIIdentityCredentialStorageService.idl	Thu Jun 05 00:19:41 2025 +0300
@@ -31,6 +31,22 @@
     // Delete the entry for all tuples (rpPrincipal, idpPrincipal, *).
     void disconnect(in nsIPrincipal rpPrincipal, in nsIPrincipal idpPrincipal);
 
+     // Store a "lightweight" identity credential
+     [noscript]
+    void storeIdentityCredential(in IPCIdentityCredentialRef credential);
+
+    // Retrieve all "lightweight" identity credentials stored by these IDPs
+     [noscript]
+    Array<IPCIdentityCredential> getIdentityCredentials(in Array<nsIPrincipal> idpPrincipal);
+
+    // Retrieve all "lightweight" identity credentials stored with this type, for the provided RP
+     [noscript]
+    Array<IPCIdentityCredential> getIdentityCredentialsOfType(in ACString type);
+
+    // Delete a  "lightweight" identity credential
+     [noscript]
+    void deleteIdentityCredential(in IPCIdentityCredentialRef credential);
+
     // Delete all data in this service.
     void clear();
 